src/pages/dashboard/Dashboard.js

```diff
@@ -20,15 +21,135 @@ const DashboardDiv = styled.div`
 `
 
 
-function Dashboard (props) {
-return (
+class Dashboard extends React.Component {
+
+  state = {
+    adminName: '',
+    condoName: '',
+    condoAddress: '',
+    
+    unitName: '',
+    
+    resName: '',
+    resLastname: '',
+    resIdNumber: '',
+    resPhone: '',
+    resEmail: '',
+    resPassword: '',
+    resUnit: '',
+
+    adminid: '',
+    condoid: '',
+    message: '',
+  } 
+  
+  async componentDidMount() {
+    try {
+      const token = localStorage.getItem('token')
+      const { data } = await axios({
+        method: 'GET',
+        baseURL: 'http://localhost:8000',
+        url: '/admin',
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+
+      this.setState({ adminName: data.name, adminid: data.id })
+
+    } catch(err) {
+      
+      localStorage.removeItem('token')
+      this.props.history.push('/login')
+      
+    }
+  }
+  handleChange = (e) => {
+
+    e.preventDefault()
+
+    const { name, value } = e.target
+
+    this.setState({ [name]: value })
+  }
+  addToDatabase = (endpoint, statePart) => async (e) => {
+    
+    e.preventDefault()
+    let toPost = {}
+    
+    switch (endpoint) {
+      case 'condo':
+        const { condoName , condoAddress } = this.state
+        toPost = {
+          ...statePart,
+          name: condoName,
+          address: condoAddress,
+        }
+        break;
+      case 'unit':
+        const { unitName, condoid } = this.state
+        toPost = {
+          ...statePart,
+          name: unitName,
+          condoId: condoid
+        }
+        break;
+      case 'resident':
+        const { resName, resLastname, resIdNumber, resPhone, resEmail, resPassword, resUnit } = this.state
+        toPost = {
+          ...statePart,
+          name: resName,
+          lastName: resLastname,
+          idNumber: resIdNumber,
+          phone: resPhone,
+          email: resEmail,
+          password: resPassword,
+          unitId: resUnit
+        }
+        break;
+    
+      default:
+        break;
+    }
+
+    try {
+      const token = localStorage.getItem('token')
+      const { data, message } = await axios({
+        method: 'POST',
+        baseURL: 'http://localhost:8000',
+        url: `/${endpoint}`,
+        data: toPost,
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+      if (endpoint === 'condo') {
+        this.setState({...this.state, condoid: data.data._id, message: message})
+      }
```

Comment:
Acá debería haber un else, los 2 this.setState se pueden ejecutar a la vez

Reply:
Tienes razon